gradle-plugin/src/main/kotlin/xyz/block/artifactswap/ArtifactSwapSettingsPlugin.kt

```diff
@@ -130,11 +134,19 @@ class ArtifactSwapSettingsPlugin : Plugin<Settings> {
     }
   }
 
-  /**
-   * Applies the publish plugin to all projects when sandbag publishing is enabled.
-   */
+  /** Applies the publish plugin to all projects when artifact publishing is enabled. */
   private fun Settings.maybeApplyPublishPlugin() {
-    if (isArtifactPublishingEnabled) {
+    gradle.settingsEvaluated {
+      logger.lifecycle("publish enabled ${extension.publishing.enabled.get()}")
+      if (extension.publishing.enabled.get()) {
```

Comment:
Why is it important to evaluate that property here? (I'm just always suspicious of things that "look like" afterEvaluate. Is this preferable to checking the final value of `enabled` in each project context?

Reply:
the values from extensions are never available in the initial plugin application, you can only read them in afterSettings (or afterEvaluate for plugins)